design_drafts/controller_execution_management.md

```diff
@@ -0,0 +1,125 @@
+# Controller Execution Management
+
+The following shall discuss the execution of loaded controllers inside the `ControllerManager`.
+As of today, the loaded controllers are processed in a sequential order which doesn't allow for much flexibility or interoperability between the controllers.
+We identified a few essential criteria when it comes to the execution of controllers:
+
+* Sequential Execution of Controllers
+* Parallel Execution of Controllers
+* Nesting Controllers
+* Variable Frequencies of Controllers
+
+We'll go into more details below for each aspect.
+
+
+## Prerequisites
+
+### Flexible Joint State Message
+
+Over the course of this document, we are referring to a "Flexible Joint State Message".
+This data type is precisely described in [here](flexible_joint_states_msg.md).
+
+What is important to understand for this article is that this joint state message serves as a general container structure for various, dynamically specified key values pairs.
+Every controller can hereby access values which are specific to itself without being limited to a set of pre-defined keys.
```

Comment:
Do we need to explain how the controllers have access to them here? (ex. IPC, messages...?)

Reply:
I consider the article about the `flexible joint state message` currently as incomplete and would prefer having that sort of discussion in a new PR for that article.
But in principle, I think it should stay somehow opaque to the controller what sort of message is present and how the input/output keys are connected. I think it's sufficient for an individual controller to read and write custom fields in this message. Similarly, they should not be concerned with access control in a sense that each controller has full control over the associated keys.
Does this make sense?